salt/_modules/metalk8s_solutions.py

```diff
@@ -1,186 +1,122 @@
-    )
-    api_instance = kubernetes.client.CoreV1Api()
-
-    # Retrieve the existing ConfigMap
-    configmap = __salt__['metalk8s_kubernetes.show_configmap'](
-        SOLUTIONS_CONFIG_MAP,
-        namespace=SOLUTIONS_CONFIG_MAP_NAMESPACE,
-        context=context,
-        kubeconfig=kubeconfig,
-    )
-    if not configmap:
-        log.exception(
-            'Cannot unregister Solution: ConfigMap "%s" is missing.',
-            SOLUTIONS_CONFIG_MAP
-        )
+    Any ISO9660 mounted in `/srv/scality` that isn't for MetalK8s is considered
+    to be a Solution archive.
+    """
+    mountpoint, mount_info = mount_tuple
+
+    if not mountpoint.startswith('/srv/scality/'):
         return False
 
-    old_versions = json.loads((configmap['data'] or {}).get(name, '[]'))
-    new_versions = [
-        version_dict for version_dict in old_versions
-        if version_dict['version'] != version
-    ]
-    # If this is the last registered version then remove all the entry
-    if new_versions == []:
-        del configmap['data'][name]
-        # Patching a CM while removing a key does not work, we need to replace it
-        try:
-            api_instance.replace_namespaced_config_map(
-                SOLUTIONS_CONFIG_MAP,
-                SOLUTIONS_CONFIG_MAP_NAMESPACE,
-                configmap
-            )
-        except ApiException as exc:
-            log.exception('Failed to patch ConfigMap "%s": %s',
-                          SOLUTIONS_CONFIG_MAP, str(exc))
-            return False
-    else:
-        body = {'data': {name: json.dumps(new_versions)}}
-        try:
-            api_instance.patch_namespaced_config_map(
-                SOLUTIONS_CONFIG_MAP,
-                SOLUTIONS_CONFIG_MAP_NAMESPACE,
-                body
-            )
-        except ApiException as exc:
-            log.exception('Failed to patch ConfigMap "%s": %s',
-                          SOLUTIONS_CONFIG_MAP, str(exc))
-            return False
+    if mountpoint.startswith('/srv/scality/metalk8s-'):
+        return False
+
+    if mount_info['fstype'] != 'iso9660':
+        return False
 
     return True
+
+
+def list_available():
+    """Get a view of mounted Solution archives.
+
+    Result is in the shape of a dict, with Solution names as keys, and lists
+    of mounted archives (each being a dict of various info) as values.
+    """
+    result = collections.defaultdict(list)
+
+    active_mounts = __salt__['mount.active']()
+
+    solution_mounts = filter(_is_solution_mount, active_mounts.items())
+
+    for mountpoint, mount_info in solution_mounts:
+        solution_info = __salt__['metalk8s.archive_info_from_iso'](
+            mount_info['alt_device']
+        )
+        machine_name = solution_info['name'].replace(' ', '-').lower()
+        version = solution_info['version']
+
+        result[machine_name].append({
+            'display_name': solution_info['name'],
+            'machine_id': '{}-{}'.format(machine_name, version),
```

Comment:
`solution_id` may be better no ? Because nothing really related to "machine" here right ?

Reply:
The point is this name/ID is for machine use, while `display_name` is intended for humans. Maybe I can add a comment to explicit this distinction.

Reply:
Oh okay machine in that sense

Reply:
Maybe simply call them `name` and `id`, as `name` are usually for human whereas `id` are usually for computer.